ogr/ogrsf_frmts/openfilegdb/ogropenfilegdbdatasource_write.cpp

```diff
@@ -1669,6 +1670,191 @@ const GDALRelationship* OGROpenFileGDBDataSource::GetRelationship(const std::str
     return it->second.get();
 }
 
+
+/************************************************************************/
+/*                          AddRelationship()                           */
+/************************************************************************/
+
+bool OGROpenFileGDBDataSource::AddRelationship(std::unique_ptr<GDALRelationship>&& relationship,
+                                               std::string& failureReason)
+{
+    const auto relationshipName = relationship->GetName();
```

Comment:
I guess there must be restrictions on the validity of relationship names for geodatabases. Perhaps similar to those of tables ?